processes/WIP-bug_bounty_proposal.md

```diff
@@ -0,0 +1,71 @@
+
+1. Improved Security Depth of Coverage • Finding security defects is specialized work that benefits from deep expertise and fresh eyes
+1. Validate Nodejs as a secure platform • Expanding the community into the security world is a growth area and enables new audiences for nodejs to be proposed
+1. Improved Security Scope of Coverage • Reduced Risk we have bugs that are undiscovered currently exposing ourselves and our users to unnecessary risk
+1. Improved Process in SDLC • Ongoing analysis of discoveries provides an opportunity to improve security through practices and/or security focus and tooling.
+
+## Proposal Process
+
+The proposed process for considering a Bug Bounty program makes heavy reference to existing work that the Security WG has already started (establishing a triage team, organized disclosure system, etc.) and ongoing partnership with HackerOne. High level process:
+
+-[ ] Security WG reviews readiness and desire to have such a program (presume yes unless someone objects)
+-[ ] Security WG presents proposal to CTC (see next steps on how to get there)
+-[ ] Security WG & CTC establish Bug Bounty Leader & Bug Bounty Team (will take additional coordination across org)
+
+## Next steps
+
+1. Core team readiness review and architectural decisions
+
+   - goals/kpi (scope, responsiveness goals, impact of defects, team size, budget)
+   - private vs. public
+   - existing security analysis
+   - backlog validation
+
+1. Engage other committies to let them know of the proposal (community committee as there is a PR element, others?)
+1. Review and evolve draft note into a proposal
+1. Create draft Security Page updates that describe the rules of the road
+
+## Readiness review and questions for input
+
+As preparation, the Security WG already has an organization within HackerOne and the general framework as well as the technical tools to offer a Bug Bounty program.
+
+1. Do we have a security assessment of the current node.js core? *no formal assessment, should not block, but increases risk of public bounty program*
+1. Do we have any scope limitations on what we would declare? *yes, we do, just what are they exactly?*
+1. Should we bounty "LTS" version bugs and non-LTS the same *(gut is yes)*? Release vs. HEAD *(gut is no)*?
+1. Do we have enough volunteer staff (need to get HackerOne help to know what the right number might should be)
+1. Would a private bounty program be acceptable to start? *Lots of indications from those who have gone before that we should start a closed / invite program first*
+1. Does the security-wg or another group own the communication plan for describing results to the broader community?
+1. Do we know where we will get sufficient funding sources? *HackerOne I think has suggested they can help with some elements and that there are funds that can be tapped, but I don't have a sense of scope still.*
```

Comment:
I propose this to include high-impact modules on NPM (> N million downloads per month).

Reply:
The Internet Bug Bounty (IBB) is only offering to support Node.js core at this time, but supporting high-impact modules would be a great use of Node.js Foundation (or other sources) funding. However, I highly recommend starting small and slow. Perhaps only start with Node.js core and expand from there?

Reply:
I think the rep for HackerOne said it would be "free" as they would provide the required funding.

Reply:
Just to clarify what the IBB is offering is slightly different than most bug bounties in that it really is a bounty covering issues that have already been publicly disclosed and fixed. There's no obligation for Node.js to actually use HackerOne for their incoming reports, though of course, the offer is there (and free to use). In the case of the IBB, we would just take reports from researchers who have already worked with you in whatever VRP process you have to get their report fixed and disclosed, and then we would offer a bounty afterwards based on the severity of the issue. However, if Node.js uses HackerOne for handling incoming issues, then it would all just work as a normal bug bounty program, and Node.js folks would be empowered to pay out bounties based on the pre-approved bounty table (based on severity of the issue).

Reply:
@reedloden I propose that should use HackerOne to process reports.

I understand the use is not required, but I see no benefit in avoiding the use of the platform.

Reply:
Sounds good to me! I agree that works better for all, but just making sure you know the option is there. :)